test/e2e/main.go

```diff
@@ -0,0 +1,993 @@
+			},
+			InstanceType: "m4.xlarge",
+			Placement: clusterapiaproviderawsv1alpha1.Placement{
+				Region:           "us-east-1",
+				AvailabilityZone: "us-east-1a",
+			},
+			Subnet: clusterapiaproviderawsv1alpha1.AWSResourceReference{
+				Filters: []clusterapiaproviderawsv1alpha1.Filter{
+					{
+						Name:   "tag:Name",
+						Values: []string{fmt.Sprintf("%s-worker-*", clusterID)},
+					},
+				},
+			},
+			IAMInstanceProfile: &clusterapiaproviderawsv1alpha1.AWSResourceReference{
+				ID: &iamInstanceProfileID,
+			},
+			KeyName: &keyName,
+			Tags: []clusterapiaproviderawsv1alpha1.TagSpecification{
+				{
+					Name:  "openshift-node-group-config",
+					Value: "node-config-master",
+				},
+				{
+					Name:  "host-type",
+					Value: "master",
+				},
+				{
+					Name:  "sub-host-type",
+					Value: "default",
+				},
+			},
+			SecurityGroups: []clusterapiaproviderawsv1alpha1.AWSResourceReference{
+				{
+					Filters: []clusterapiaproviderawsv1alpha1.Filter{
+						{
+							Name:   "tag:Name",
+							Values: []string{fmt.Sprintf("%s-*", clusterID)},
+						},
+					},
+				},
+			},
+			PublicIP: &publicIP,
+		}
+
+		m := &clusterv1alpha1.Machine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:         machineName,
+				Namespace:    testNamespace,
+				GenerateName: "vs-master-",
+				Labels: map[string]string{
+					"sigs.k8s.io/cluster-api-cluster":      "tb-asg-35",
+					"sigs.k8s.io/cluster-api-machine-role": "infra",
+					"sigs.k8s.io/cluster-api-machine-type": "master",
+				},
+			},
+			Spec: clusterv1alpha1.MachineSpec{
+				ProviderConfig: clusterv1alpha1.ProviderConfig{
+					Value: &runtime.RawExtension{
+						Object: a,
+					},
+				},
+				Versions: clusterv1alpha1.MachineVersionInfo{
+					Kubelet:      "1.10.1",
+					ControlPlane: "1.10.1",
+				},
+			},
+		}
+
+		if err := createMachine(testConfig, m); err != nil {
+			return err
+		}
+
+		// Verify cluster and machine have been deployed
+		var cluster, machine bool
+		err = wait.Poll(pollInterval, timeoutPoolMachineRunningInterval, func() (bool, error) {
+			if _, err := testConfig.CAPIClient.ClusterV1alpha1().Clusters(testNamespace).Get(clusterID, metav1.GetOptions{}); err == nil {
+				cluster = true
+				log.Info("Cluster object has been created")
+			}
+
+			if _, err := testConfig.CAPIClient.ClusterV1alpha1().Machines(testNamespace).Get(machineName, metav1.GetOptions{}); err == nil {
+				machine = true
+				log.Info("Machine object has been created")
+			}
+
+			if cluster && machine {
+				return true, nil
+			}
+			log.Info("Waiting for cluster and machine to be created")
+			return false, nil
+		})
+
+		if err != nil {
+			return err
+		}
+
+		log.Info("The cluster and the machine have been deployed")
+
+		err = wait.Poll(pollInterval, timeoutPoolAWSInterval, func() (bool, error) {
+			log.Info("Waiting for aws instances to come up")
+			runningInstances, err := testConfig.AWSClient.GetRunningInstances(clusterID)
```

Comment:
This is ok to begging with, then we'll want to declare specific cases/validations e.g:
for manifest-1 I expect this sg and this subnet, the likes of:
```
	testCases := []struct {
		manifest       string
		expectedSG     string
		expectedSubnet
	}{
		{
			manifest:       "machineSet-1.yaml,
			expectedSG:     "sg1, sg2",
			expectedSubnet: "subnet1, subnet2",
		},
		{
			manifest:       "machineSet-2.yaml,
			expectedSG:     "sg1, sg2",
			expectedSubnet: "subnet1, subnet2",
		},
	}
```